sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/FileStatusCache.scala

```diff
@@ -36,6 +37,8 @@ import org.apache.spark.util.SizeEstimator
 object FileStatusCache {
   private var sharedCache: SharedInMemoryCache = _
 
+  private[spark] val sessionToCache = new mutable.HashMap[String, FileStatusCache]
```

Comment:
Can this itself grow large? I'm also wary of maps that track stuff over long periods as a potential memory problem

Reply:
@srowen Thanks for reminding, I should remove the entry from the map when closing a session, I will fix it.
And with this fix, I think the memory problem should be fine, because the total size of `sessionToCache` is consist of 'SessionUUID' and `FileStatusCache`, the total size of `FileStatusCache` is controlled by the max size of `sharedCache`, and the remaining parts is equal to (the number of SparkSession * the size of every `SessionUUID`), this would not cost too much memory.